### ISSUE TITLE
.NET 6 with Android cannot restore packages properly

### ISSUE THREAD

**Issue by mattleibow** [NuGet/Home#10889]
## Details about Problem

I have a NuGet package that contains things for net6.0-android, monoandroid and netstandard. However, the restore process has some weird happenings:
 - it uses the netstandard dlls
 - it uses the monoandroid targets
 - it uses the runtimes/android/native .so files

Basically it is all weird. I tried to reduce the package to just net6.0 and net6.0-android, but it still picks net6.0. In fact, the only way to make it pick android is that the project needs to be android and the nuget needs to have nothing less specific.

NuGet product used (NuGet.exe | Visual Studio | MSBuild.exe | dotnet.exe): MSBuild

Product version: 16.10.0+4242f381a

Worked before? If so, with which NuGet version: new things

## Repro steps and/or sample project

[project.assets.json](https://gist.github.com/mattleibow/4b637df71be4f2ac0c44163e4bce0dec)
[HarfBuzzSharp.2.6.1.8-pr.1707.18.nupkg.zip](https://github.com/NuGet/Home/files/6532226/HarfBuzzSharp.2.6.1.8-pr.1707.18.nupkg.zip)

`<TargetFrameworks>net6.0-android</TargetFrameworks>`

### Verbose Logs

[msbuild.binlog.zip](https://github.com/NuGet/Home/files/6532212/msbuild.binlog.zip)

**Comment by mattleibow** [comment]
I see this was merged: https://github.com/NuGet/NuGet.Client/pull/4042

Not sure where that live now with regards to bits and bytes in the wild.

**Comment by mattleibow** [comment]
Also this is related: https://github.com/xamarin/xamarin-android/issues/5896

**Comment by mattleibow** [comment]
~~I managed to work around this with:~~

**EDIT**
This breaks in the IDE as this runs on restore, but seems the IDE does something slightly different.

```xml
  <Target Name="_CollectPackageReferencesFixes" BeforeTargets="CollectPackageReferences">
    <PropertyGroup Condition="'$(PackageTargetFallback)' != '' and '$(AssetTargetFallback)' == ''">
      <AssetTargetFallback>$(PackageTargetFallback)</AssetTargetFallback>
      <PackageTargetFallback></PackageTargetFallback>
    </PropertyGroup>
  </Target>
```

**Comment by zkat** [comment]
@mattleibow we don't have the bits from https://github.com/NuGet/NuGet.Client/pull/4042 included in 16.10. I'm curious whether that PR is gonna fix this issue altogether.

**Comment by zkat** [comment]
This should be fixed now. Please feel free to ping me/reopen if it's not working due to some issue on our end!
